test/modules/carto/api/maps-api-client.spec.js

```diff
@@ -0,0 +1,108 @@
+import test from 'tape-catch';
+import {_getTileJSON, setDefaultCredentials} from '@deck.gl/carto';
+
+const SQL_RESPONSE = {
+  tilejson: '2.2.0',
+  tiles: [
+    'https://maps-api-v2.us.carto.com/user/public/tile/map-config/bebb71dac05816a8234ab721a6fd407a/1600089410773/{z}/{x}/{y}.mvt?api_key=default_public'
+  ],
+  tilestats: {}
+};
+
+const MAP_CONFIG = {
+  version: '2.0.0',
+  buffer_size: 16,
+  tile_extent: 4096,
+  layers: [
+    {
+      type: 'sql',
+      source: 'postgres',
+      options: {sql: 'SELECT * FROM ne_10m_airports', vector_extent: 4096}
+    }
+  ]
+};
+
+const TEST_CASES = [
+  {
+    server: {
+      ok: false,
+      status: 401
+    },
+    credentials: {
+      username: 'publi',
+      apiKey: 'default_public'
+    }
+  },
+  {
+    server: {
+      ok: false,
+      status: 403
+    },
+    credentials: {
+      apiKey: 'defaultpublic'
+    }
+  },
+  {
+    server: {
+      ok: false,
+      status: 'unknown'
+    }
+  },
+  {
+    server: {
+      ok: true
+    }
+  }
+];
+
+const VERSION_TEST_CASES = ['2', 'v3', 3];
+
+test('mapApiClient#getTileJSON', async t => {
+  /* global global, window */
+  const _global = typeof global !== 'undefined' ? global : window;
+  const fetch = _global.fetch;
+
+  for (const tc of TEST_CASES) {
+    _global.fetch = () => {
+      return new Promise(resolve => {
+        return resolve({
+          json: () => SQL_RESPONSE,
+          ...tc.server
+        });
+      });
+    };
+
+    if (!tc.server.ok) {
+      setDefaultCredentials(tc.credentials);
+
+      try {
+        await _getTileJSON({mapConfig: MAP_CONFIG});
+      } catch (err) {
+        t.pass(`Correctly throws ${err}`);
+      }
+      /* eslint-disable-next-line no-continue */
+      continue;
+    }
+
+    const func = await _getTileJSON({mapConfig: MAP_CONFIG});
+    t.ok(func.tiles[0] === SQL_RESPONSE.tiles[0], 'getTileJSON correctly returns tiles endpoint');
+  }
+
+  for (const tc of VERSION_TEST_CASES) {
```

Comment:
This should be probably split into a different test, as this one is already quite big @AdriSolid

Reply:
Will open a new PR to split the test :+1: